include/bluetooth/mesh/time_srv.h

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_time_srv Time Server model
+ * @{
+ * @brief API for the Time Server model.
+ */
+
+#ifndef BT_MESH_TIME_SRV_H__
+#define BT_MESH_TIME_SRV_H__
+
+#include <bluetooth/mesh/mesh_time.h>
+#include <bluetooth/mesh/model_types.h>
+#include <time.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct bt_mesh_time_srv;
+struct tm;
+
+/** @def BT_MESH_TIME_SRV_INIT
+ *
+ * @brief Initialization parameters for a @ref bt_mesh_time_srv instance.
+ *
+ * @param[in] _time_update_cb Time update callback (optional).
+ */
+#define BT_MESH_TIME_SRV_INIT(_time_update_cb)                                 \
+	{                                                                      \
+		.time_update_cb = _time_update_cb,                             \
+		.pub = { .update = _bt_mesh_time_srv_update_handler,           \
+			 .msg = NET_BUF_SIMPLE(BT_MESH_MODEL_BUF_LEN(          \
+				 BT_MESH_TIME_OP_TIME_SET,                     \
+				 BT_MESH_TIME_MSG_LEN_TIME_SET)) },            \
+	}
+
+/** @def BT_MESH_MODEL_TIME_SRV
+ *
+ * @brief Time Server model composition data entry.
+ *
+ * @param[in] _srv Pointer to a @ref bt_mesh_time_srv instance.
+ */
+#define BT_MESH_MODEL_TIME_SRV(_srv)                                           \
+	BT_MESH_MODEL_CB(                                                      \
+		BT_MESH_MODEL_ID_TIME_SRV, _bt_mesh_time_srv_op, &(_srv)->pub, \
+		BT_MESH_MODEL_USER_DATA(struct bt_mesh_time_srv, _srv),        \
+		&_bt_mesh_time_srv_cb),                                        \
+	BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_TIME_SETUP_SRV,                      \
+		_bt_mesh_time_setup_srv_op,                                    \
+		&(_srv)->setup_pub,                                            \
+		BT_MESH_MODEL_USER_DATA(                                       \
+		struct bt_mesh_time_srv, _srv),                                \
+		NULL)
+
+/** Time srv update types */
+enum bt_mesh_time_update_types {
+	/** Status update */
+	BT_MESH_TIME_SRV_STATUS_UPDATE,
+	/** Set update */
+	BT_MESH_TIME_SRV_SET_UPDATE,
+	/** Zone update */
+	BT_MESH_TIME_SRV_ZONE_UPDATE,
+	/** UTC update */
+	BT_MESH_TIME_SRV_UTC_UPDATE,
+};
+
+/** Internal data structure of the Time Server instance. */
+struct bt_mesh_time_srv_data {
+	/** State at last sync point */
+	struct {
+		/** Device uptime at last sync point */
+		s64_t uptime;
+		/** Time status at last sync point */
+		struct bt_mesh_time_status status;
+	} sync;
+	/** Zone change update context */
+	struct bt_mesh_time_zone_change time_zone_change;
+	/** UTC-delta change update context */
+	struct bt_mesh_time_tai_utc_change tai_utc_change;
+	/** The Time Role of the Server instance*/
+	enum bt_mesh_time_role role;
+};
+
+/**
```

Comment:
Start doc on first line, Since the line become too long maybe split it and add brief?

```
/** @brief Time Server instance.
 *
 *  Should be initialized with @ref BT_MESH_TIME_SRV_INIT.
 */
```